docs/DeveloperGuide.md

```diff
@@ -6,33 +6,230 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### The Overall UML Diagram:
 
+![UML Diagram](assets/images/mathGenius.png)    
```

Comment:
Should you have everything in this UML diagram? Perhaps could reduce number of methods and members listed to make a summary UML diagram